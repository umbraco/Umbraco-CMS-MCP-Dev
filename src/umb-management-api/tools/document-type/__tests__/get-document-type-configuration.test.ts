import GetDocumentTypeConfigurationTool from "../get/get-document-type-configuration.js";
import { jest } from "@jest/globals";

describe("get-document-type-configuration", () => {
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(() => {
    console.error = originalConsoleError;
  });

  it("should get the global document type configuration", async () => {
    const result = await GetDocumentTypeConfigurationTool().handler({}, { signal: new AbortController().signal });
    const config = JSON.parse(result.content[0].text as string);
    expect(config).toMatchSnapshot();
  });
}); 