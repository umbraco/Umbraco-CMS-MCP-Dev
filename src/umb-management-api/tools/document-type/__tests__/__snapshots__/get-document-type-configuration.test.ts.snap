// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-document-type-configuration should get the global document type configuration 1`] = `
{
  "dataTypesCanBeChanged": "True",
  "disableTemplates": false,
  "reservedFieldNames": [
    "ContentType",
    "Key",
    "Properties",
    "Id",
    "Name",
    "UrlSegment",
    "SortOrder",
    "Level",
    "Path",
    "TemplateId",
    "CreatorId",
    "CreateDate",
    "WriterId",
    "UpdateDate",
    "Cultures",
    "ItemType",
    "Parent",
    "Children",
    "get_ContentType",
    "get_Key",
    "get_Properties",
    "GetProperty",
    "get_Id",
    "get_Name",
    "get_UrlSegment",
    "get_SortOrder",
    "get_Level",
    "get_Path",
    "get_TemplateId",
    "get_CreatorId",
    "get_CreateDate",
    "get_WriterId",
    "get_UpdateDate",
    "get_Cultures",
    "get_ItemType",
    "get_Parent",
    "IsDraft",
    "IsPublished",
    "get_Children",
  ],
  "useSegments": false,
}
`;
